src/main/scala/User.scala

```diff
@@ -1,3 +1,120 @@
-object User {
+import DataBaseConn.{getAll, putUser}
+import User.registerAndLoginRoute
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.{ContentTypes, HttpEntity, StatusCodes}
+import akka.http.scaladsl.server.Directives._
+import spray.json._
 
+import scala.concurrent.duration.DurationInt
+import scala.language.postfixOps
+import scala.util.{Failure, Success}
+
+/*
+  POST /register -> register User(username, password)
+  GET /users -> get all the users
+  GET /user/{id} -> get a user by id
+  GET /login -> OK (Successful), NotFound (Error)
+ */
+
+// id is auto incremented
+case class UserComplete(id: Int, username: String, password: String)
+case class UserPartial(username: String, password: String)
```

Comment:
This could be simply, `User`.

Also, we can have a class hierarchy here, with a User(username, password) and a UserRecord that will inherit User class.